<details>
<summary>Pre-Commit Test failed! Click here</summary>

```
[INFO] Initializing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Initializing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Installing environment for git://github.com/Lucas-C/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for git://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/adrienverge/yamllint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Tabs remover.............................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for merge conflicts................................................Passed
Fix End of Files.........................................................Passed
Check for added large files..............................................Passed
Check for case conflicts.................................................Passed
Check JSON...........................................(no files to check)Skipped
Check for broken symlinks............................(no files to check)Skipped
Detect Private Key.......................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
yamllint.................................................................Failed
- hook id: yamllint
- exit code: 1

manifests/base/deployments/argocd-dex-server_patch.yaml
  20:9      error    wrong indentation: expected 6 but found 8  (indentation)
```
</details>

--- a/manifests/base/deployments/argocd-application-controller_patch.yaml
+++ b/manifests/base/deployments/argocd-application-controller_patch.yaml
@@ -1,0 +1,12 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: argocd-application-controller
+  annotations:
+    alpha.image.policy.openshift.io/resolve-names: '*'
+spec:
+  template:
+    spec:
+      containers:
+      - image: argocd-custom
+        name: argocd-application-controller

--- a/manifests/base/deployments/argocd-dex-server_patch.yaml
+++ b/manifests/base/deployments/argocd-dex-server_patch.yaml
@@ -2,14 +2,20 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: argocd-dex-server
+  annotations:
+    alpha.image.policy.openshift.io/resolve-names: '*'
 spec:
   template:
     spec:
       containers:
       - name: dex
+        image: argocd-dex-server
         env:
         - name: DEX_OAUTH_CLIENT_SECRET
           valueFrom:
             secretKeyRef:
               name: argocd-dex-server-oauth-token
               key: token
+      initContainers:
+      - name: copyutil
+        image: argocd-custom

--- a/manifests/base/deployments/argocd-redis_patch.yaml
+++ b/manifests/base/deployments/argocd-redis_patch.yaml
@@ -6,6 +6,8 @@ metadata:
     app.kubernetes.io/name: argocd-redis
     app.kubernetes.io/part-of: argocd
   name: argocd-redis
+  annotations:
+    alpha.image.policy.openshift.io/resolve-names: '*'
 spec:
   selector:
     matchLabels:
@@ -21,7 +23,7 @@ spec:
         - ""
         - --appendonly
         - "no"
-        image: redis:5.0.8
+        image: redis
         imagePullPolicy: Always
         name: redis
         ports:

--- a/manifests/base/deployments/argocd-repo-server_patch.yaml
+++ b/manifests/base/deployments/argocd-repo-server_patch.yaml
@@ -2,12 +2,14 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: argocd-repo-server
+  annotations:
+    alpha.image.policy.openshift.io/resolve-names: '*'
 spec:
   template:
     spec:
       containers:
       - name: argocd-repo-server
-        image: quay.io/aicoe/argocd:v1.7.4-3
+        image: argocd-custom
         env:
         - name: GNUPGHOME
           value: /home/argocd/.gnupg

--- a/manifests/base/deployments/argocd-server_patch.yaml
+++ b/manifests/base/deployments/argocd-server_patch.yaml
@@ -2,11 +2,14 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: argocd-server
+  annotations:
+    alpha.image.policy.openshift.io/resolve-names: '*'
 spec:
   template:
     spec:
       containers:
       - name: argocd-server
+        image: argocd-custom
         command:
         - argocd-server
         - --staticassets

--- a/manifests/base/imagestreams/argocd-custom.yaml
+++ b/manifests/base/imagestreams/argocd-custom.yaml
@@ -1,0 +1,14 @@
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: argocd-custom
+spec:
+  tags:
+    - name: latest
+      from:
+        kind: DockerImage
+        name: quay.io/aicoe/argocd:v1.7.4-3
+      importPolicy: {}
+      referencePolicy:
+        type: Local

--- a/manifests/base/imagestreams/argocd-dex-server.yaml
+++ b/manifests/base/imagestreams/argocd-dex-server.yaml
@@ -1,0 +1,14 @@
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: argocd-dex-server
+spec:
+  tags:
+    - name: latest
+      from:
+        kind: DockerImage
+        name: quay.io/dexidp/dex:v2.22.0
+      importPolicy: {}
+      referencePolicy:
+        type: Local

--- a/manifests/base/imagestreams/redis.yaml
+++ b/manifests/base/imagestreams/redis.yaml
@@ -1,0 +1,14 @@
+---
+apiVersion: image.openshift.io/v1
+kind: ImageStream
+metadata:
+  name: redis
+spec:
+  tags:
+    - name: latest
+      from:
+        kind: DockerImage
+        name: redis:5.0.8
+      importPolicy: {}
+      referencePolicy:
+        type: Local

--- a/manifests/base/kustomization.yaml
+++ b/manifests/base/kustomization.yaml
@@ -4,6 +4,9 @@ resources:
 - github.com/argoproj/argo-cd/manifests/base?ref=v1.7.4
 - routes/argocd_server_route.yaml
 - serviceaccounts/argocd-manager.yaml
+- imagestreams/argocd-custom.yaml
+- imagestreams/argocd-dex-server.yaml
+- imagestreams/redis.yaml
 patchesStrategicMerge:
 - services/argocd_metrics_service_patch.yaml
 - services/argocd_server_metrics_service_patch.yaml
@@ -11,3 +14,4 @@ patchesStrategicMerge:
 - deployments/argocd-server_patch.yaml
 - deployments/argocd-redis_patch.yaml
 - deployments/argocd-dex-server_patch.yaml
+- deployments/argocd-application-controller_patch.yaml

